tracing-subscriber/src/layer/context.rs

```diff
@@ -0,0 +1,473 @@
+use tracing_core::{metadata::Metadata, span, subscriber::Subscriber, Event};
+
+#[cfg(feature = "registry")]
+use crate::registry::Registry;
+use crate::{
+    filter::FilterId,
+    registry::{self, LookupSpan, SpanRef},
+};
+/// Represents information about the current context provided to [`Layer`]s by the
+/// wrapped [`Subscriber`].
+///
+/// To access [stored data] keyed by a span ID, implementors of the `Layer`
+/// trait should ensure that the `Subscriber` type parameter is *also* bound by the
+/// [`LookupSpan`]:
+///
+/// ```rust
+/// use tracing::Subscriber;
+/// use tracing_subscriber::{Layer, registry::LookupSpan};
+///
+/// pub struct MyLayer;
+///
+/// impl<S> Layer<S> for MyLayer
+/// where
+///     S: Subscriber + for<'a> LookupSpan<'a>,
+/// {
+///     // ...
+/// }
+/// ```
+///
+/// [`Layer`]: ../layer/trait.Layer.html
+/// [`Subscriber`]: https://docs.rs/tracing-core/latest/tracing_core/trait.Subscriber.html
+/// [stored data]: ../registry/struct.SpanRef.html
+/// [`LookupSpan`]: "../registry/trait.LookupSpan.html
+#[derive(Debug)]
+pub struct Context<'a, S> {
+    subscriber: Option<&'a S>,
+    filter: FilterId,
+}
```

Comment:
...why is this considered new? i thought there's already a [Context](https://docs.rs/tracing-subscriber/0.2.20/tracing_subscriber/layer/struct.Context.html)?

Reply:
i moved it to a new file, since the `layer.rs` file was getting very large. i added a note about this in the PR description.

Reply:
ah, missed that. thanks.